### ISSUE TITLE
[BUG] Connection refused 127.0.0.1:8080 since 1.1.7-4

### ISSUE THREAD

**Issue by salavessa** [graphite-project/docker-graphite-statsd#149]
**Describe the bug**
1.1.7-4 is failing with: `[tagdb] Error tagging <some.tag>: Error requesting http://127.0.0.1:8080/tags/tagMultiSeries: HTTPConnectionPool(host='127.0.0.1', port=8080): Max retries exceeded with url: /tags/tagMultiSeries (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f58783c0190>: Failed to establish a new connection: [Errno 111] Connection refused'))`

**To Reproduce**
Run with docker image 1.1.7-4 (updating from 1.1.7-3)

**Expected behavior**
Error not showing and indexing/querying to work.

**Environment (please complete the following information):**
 - OS flavor: AWS AL2 (4.14.181-142.260.amzn2.x86_64)
 - Image version: 1.1.7-4
 - Custom volumes mounted: /opt/graphite/conf/storage-schemas.conf, /opt/graphite/storage, /var/log/graphite

**Additional context**
Reverting to 1.1.7-3 gets it working fine again.

Thanks!

**Comment by deniszh** [comment]
Unfortunately, I can't reproduce. 1.1.7-4 working fine with default configuration, so, it's probably depend on your environment variables or config.
Are you sure that you didn't change any config except `storage-schemas.conf`?
Are you defining any environment variables?

**Comment by dkozinn** [comment]
I've got the same issue with what appears to be a slightly older version of AWS linux: 4.14.173-137.229.amzn2.x86_64. I've got /opt/statsd/config, /opt/graphite/conf, and /opt/graphite/storage mounted.  Two environment variables are defined: 
```
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
STATSD_INTERFACE=udp
```
The only thing changed in any of the configuration files was to try to fix the error by replacing
```
GRAPHITE_URL = http://127.0.0.1:8080
```
in carbon.conf with the actual hostname (which didn't work). Prior to the update I'd been using the default.

As with  the other poster, reverting to 1.1.7-3 fixes the issue.

**Comment by salavessa** [comment]
Besides what has been reported, the only "extra" option used is a custom/user docker "bridge" network (Subnet: 172.30.64.0/18, Gateway: 172.30.64.1).
The docker command to run graphite is something like:
```
docker run -d --name graphite \
  --restart=always \
  --network=user-network \
  -p 2003-2004:2003-2004 \
  -p 2023-2024:2023-2024 \
  -p 8125:8125/udp \
  -p 8126:8126 \
  -v /local/graphite/conf/storage-schemas.conf:/opt/graphite/conf/storage-schemas.conf \
  -v /local/graphite/storage:/opt/graphite/storage \
  -v /local/graphite/log:/var/log/graphite \
  graphiteapp/graphite-statsd:1.1.7-3
```

**Comment by salavessa** [comment]
The problem seems to be that (at least) `gunicorn` doesn't start for some reason.
Doing some further checks it seems to have something to do with the existing storage folder as if I run 1.1.7-4 from a clean or not mounted storage volume then I can see `gunicorn` running.

After checking the processes both on 1.1.7-3 and 1.1.7-4 containers it's possible the issue is in the workflow to start `gunicorn` and not `gunicorn` itself.
Below are the processes non-common processes between 1.1.7-3 and 1.1.7-4 containers:
```
[1.1.7-3 ONLY]:
{gunicorn} /opt/graphite/bin/python3 /opt/graphite/bin/gunicorn wsgi --pythonpath=/opt/graphite/webapp/graphite --preload --threads=1 --worker-class=sync --workers=4 --limit-request-line=0 --max-requests=1000 --timeout=65 --bind=0.0.0.0:8080 --log-file=/var/log/gunicorn.log
{gunicorn} /opt/graphite/bin/python3 /opt/graphite/bin/gunicorn wsgi --pythonpath=/opt/graphite/webapp/graphite --preload --threads=1 --worker-class=sync --workers=4 --limit-request-line=0 --max-requests=1000 --timeout=65 --bind=0.0.0.0:8080 --log-file=/var/log/gunicorn.log
{gunicorn} /opt/graphite/bin/python3 /opt/graphite/bin/gunicorn wsgi --pythonpath=/opt/graphite/webapp/graphite --preload --threads=1 --worker-class=sync --workers=4 --limit-request-line=0 --max-requests=1000 --timeout=65 --bind=0.0.0.0:8080 --log-file=/var/log/gunicorn.log
{gunicorn} /opt/graphite/bin/python3 /opt/graphite/bin/gunicorn wsgi --pythonpath=/opt/graphite/webapp/graphite --preload --threads=1 --worker-class=sync --workers=4 --limit-request-line=0 --max-requests=1000 --timeout=65 --bind=0.0.0.0:8080 --log-file=/var/log/gunicorn.log
{gunicorn} /opt/graphite/bin/python3 /opt/graphite/bin/gunicorn wsgi --pythonpath=/opt/graphite/webapp/graphite --preload --threads=1 --worker-class=sync --workers=4 --limit-request-line=0 --max-requests=1000 --timeout=65 --bind=0.0.0.0:8080 --log-file=/var/log/gunicorn.log
```
```
[1.1.7-4 ONLY]:
{django-admin.py} /opt/graphite/bin/python3 /opt/graphite/bin/django-admin.py createsuperuser
{run} /bin/sh ./run
expect /opt/graphite/bin/django_admin_init.exp

```
Hope this helps.

**Comment by deniszh** [comment]
OK, I reproduced a bug, fix in progress. But in general, nothing really interesting happened  in 1.1.7-4, no need to upgrade of 1.1.7-3 working fine for you, btw.

**Comment by deniszh** [comment]
Fixed in 1.1.7-5
